src/main/java/io/vertx/core/http/impl/HttpUtils.java

```diff
@@ -925,23 +928,32 @@ private static int validateLatinValue(CharSequence seq, int state, byte latinCha
      * 1: The previous character was CR
      * 2: The previous character was LF
      */
-    if ((latinChar & HIGHEST_INVALID_VALUE_CHAR_MASK) == 0 || latinChar == 0x7F) {
-      validateNonPrintableCtrlLatin(seq, latinChar);
+    if (latinChar == 0x7F) {
+      throw new IllegalArgumentException("a header value contains a prohibited character '127': " + seq);
     }
-
-    // Check the CRLF (HT | SP) pattern
-    if (state == NO_CR_LF_STATE) {
-      switch (latinChar) {
-        case '\r':
-          return CR_STATE;
-        case '\n':
-          return LF_STATE;
+    if ((latinChar & HIGHEST_INVALID_VALUE_CHAR_MASK) == 0) {
+      // this is a rare scenario
+      validateNonPrintableCtrlLatin(seq, latinChar);
+      // this can include LF and CR as they are non-printable characters
+      if (state == NO_CR_LF_STATE) {
+        // Check the CRLF (HT | SP) pattern
```

Comment:
if that is rare, shouldn't we encapsulate the if block in a method to reduce method size ?